[MM-54296] Admin Reporting for Users, new User Management screen

devinbinnie opened the pull request:
#### Summary
This PR contains a redesign of the user management screen in the admin console. Changes include:
- Visual redesign of the table, implemented using `@tanstack/react-table`. The same sorting, filtering, searching, paging, and actions menu are implemented, as well as the ability to show and hide columns, and to have that data persisted between screens.
- New post stats are added to the table for PostgreSQL instances, showing each users daily activity, number of messages posted and the date of their last post, filterable by a date range.
- Added an export flow for user data, with stats filtered by date range, where an admin can start a job that when finished, will send a DM to the admin with an attached CSV report. (Note: some changes still outstanding for this to work as designed: https://github.com/mattermost/mattermost/pull/26019)

This PR contains only the front-end changes to make use of APIs and logic already available in the server. Most of the changes were reviewed between myself and @M-ZubairAhmed, so I am only asking for one dev review for the time being. Most importantly for this change will be the UX and QA reviews to ensure that everything is working as designed.

We will be following up this PR with some E2E tests to cover the general use of the admin reporting screen.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54296

#### Screenshots
![image](https://github.com/mattermost/mattermost/assets/52460000/0040138e-a2d1-48a2-b549-667eaf3d1468)
![image](https://github.com/mattermost/mattermost/assets/52460000/4c63bb69-e19f-449c-b814-ffd8f39d13ca)
![image](https://github.com/mattermost/mattermost/assets/52460000/f23066ec-d859-41d0-a46d-1d7d5944f5d7)
![image](https://github.com/mattermost/mattermost/assets/52460000/33e714b1-40dd-4290-8bab-c63352effbae)


#### Release Note
```release-note
Redesign user management screen
Add batch CSV export of users
```

mattermost-build commented:
E2E tests not automatically triggered, because the PR is not in a mergeable state. Please update the branch with the base branch and resolve outstanding conflicts.

M-ZubairAhmed commented:
/update-branch

yasserfaraazkhan commented:
/e2e-tests

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](http://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/406194) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/406194)

yasserfaraazkhan reviewed (changes requested):
wanted to share some initial observations, @devinbinnie @abhijit-singh ,

- In New UI, we can't see if the user is Deactivated, unless we filter them. 
<img width="600" alt="Screenshot 2024-01-29 at 12 48 30 PM" src="https://github.com/mattermost/mattermost/assets/15979783/d6d9bbea-abe6-4260-afad-33e3333417cc">

comparing to how it was earlier.
<img width="600" alt="image" src="https://github.com/mattermost/mattermost/assets/15979783/0eabd25b-72d9-4471-9f30-e5eb5b593b88">

- Any chance we can have a 'Reset filter' option (this option is available in other places like teams setting)
<img width="716" alt="image" src="https://github.com/mattermost/mattermost/assets/15979783/b09b6391-9f58-40d3-8d6d-98e415443e93">

- Should the Filter option be `Activated User`. there was a recent change in team & site statistics around the usage of words Active-Activated user @larkox @johndavidlugtu 

<img width="261" alt="image" src="https://github.com/mattermost/mattermost/assets/15979783/f6f5fe8b-ab7c-470d-9c61-f039c1ccc5fd">

- if term `User` is searched in console, the column header in user manager splits
<img width="560" alt="Screenshot 2024-01-29 at 1 28 28 PM" src="https://github.com/mattermost/mattermost/assets/15979783/b10ec889-d57c-4c72-90a5-87b401431d70">
 

- If we have the above search term, and apply Role filters.
<img width="736" alt="Screenshot 2024-01-29 at 1 29 05 PM" src="https://github.com/mattermost/mattermost/assets/15979783/b8becb92-308c-48a7-8c51-f98a912d75c7">

<img width="579" alt="image" src="https://github.com/mattermost/mattermost/assets/15979783/5c6f96b7-abcf-4961-8745-3b6386830fe5">


- some e2e might need locators to be updated.

- The Last post/last active, were empty. Had logged in with another user to post in channel

<img width="800" alt="image" src="https://github.com/mattermost/mattermost/assets/15979783/ae3c7cda-daf5-4c05-b388-c7e028711604">

- filters are working as expected ✅ 
- sorting, pagination ✅ 
- export data ✅ 
<img width="700" alt="image" src="https://github.com/mattermost/mattermost/assets/15979783/bb43f419-e67e-4b3e-9a8f-93040a3bdee5">

- selected Columns being displayed ✅ 
- revoke sessions ✅

devinbinnie commented:
/update-branch

larkox reviewed (changes requested):
Nothing really major, but a few comments mainly regarding style and potential improvements.

devinbinnie commented:
@yasserfaraazkhan 
> * In New UI, we can't see if the user is Deactivated, unless we filter them.

Fixed!

> * Any chance we can have a 'Reset filter' option (this option is available in other places like teams setting)

I'll leave this up to @abhijit-singh, but this would be a new ask on top. We could always add it later, so I would say we do it as a follow-up.

> * Should the Filter option be `Activated User`. there was a recent change in team & site statistics around the usage of words Active-Activated user
> * a small spelling correction needed `Manage Roles`. Also should it be like Camel casing?

Fixed the spelling, I'll leave the casing up to @abhijit-singh.

> * if term `User` is searched in console, the column header in user manager splits
> * If we have the above search term, and apply Role filters.

This is actually also causing a crash. I'll have to investigate more into how this works and how to fix it properly.

> * some e2e might need locators to be updated.
> * The Last post/last active, were empty. Had logged in with another user to post in channel

@M-ZubairAhmed has done this in another PR, I think he's going to add them here so that should be covered

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](http://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/406482) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/406482)

yasserfaraazkhan commented:
- Banner when instance is MySQL ✅ 
<img width="800" alt="image" src="https://github.com/mattermost/mattermost/assets/15979783/d0e92762-fab5-463a-83b9-85e9080c8b2b">

yasserfaraazkhan commented:
Hi @devinbinnie 
I saw couple of scenarios where UI 

1. If we De-select, all columns and just have 2 default colums to display. Navigate to Any other page in System console. Com back to the Users page. We see the colum headers as seen below.
 
<img width="700" alt="Screenshot 2024-01-31 at 12 07 22 PM" src="https://github.com/mattermost/mattermost/assets/15979783/b1c581cc-b418-4847-b642-817534c6a443">


<img width="700" alt="Screenshot 2024-01-31 at 12 02 07 PM" src="https://github.com/mattermost/mattermost/assets/15979783/03bc8624-de0f-4b83-b7e1-d95120a158f2">

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/admin_user_card/admin_user_card.tsx:
```diff
@@ -18,19 +18,37 @@ type BulletProps = {
+            <div className='AdminUserCard'>
+                <div className='AdminUserCard__header'>
+                    <ProfilePicture
+                        src=''
```
When empty it fallback to this
<img width="961" alt="Screenshot 2024-01-31 at 15 16 54" src="https://github.com/mattermost/mattermost/assets/17708702/0979cb57-93fa-45b2-87c3-0c55eb0d15b3">

larkox replied:
@abhijit-singh Do we want the bot profile picture here when a user is not found? Or do we prefer anything different?

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/list_table/list_table.tsx:
```diff
@@ -111,24 +118,28 @@ export function ListTable<TableType extends TableMandatoryTypes>(
         }
     }
 
+    const colcount = props.table.getAllColumns().length;
```
changed

yasserfaraazkhan commented:
/e2e-tests

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](http://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/406713) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/406713)

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/list_table/pagination.tsx:
```diff
@@ -6,22 +6,15 @@ import {useIntl} from 'react-intl';
-| 'isLoading'
-| 'paginationInfo'
->;
+interface Props extends Pick<TableMeta, 'onPreviousPageClick' | 'onNextPageClick' | 'disablePrevPage' | 'disableNextPage'> {
+    isLoading?: boolean;
+}
```
agreed multiline is much better in reading

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/manage_tokens_modal/__snapshots__/manage_tokens_modal.test.tsx.snap:
```diff
@@ -23,11 +23,12 @@ exports[`components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx sh
   restoreFocus={true}
   role="dialog"
-  show={false}
+  show={true}
```
yes because now its controlled by modal controller

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/list_table/list_table.tsx:
```diff
@@ -273,7 +327,7 @@ export function ListTable<TableType extends TableMandatoryTypes>(
                             value={selectedPageSize}
                             onChange={handlePageSizeChange}
-                            isDisabled={tableMeta.isLoading}
+                            isDisabled={tableMeta.loadingState === LoadingStates.Loading}
```
i dont see any drawbacks here with being verbose

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/manage_roles_modal/manage_roles_modal.tsx:
```diff
@@ -46,6 +49,7 @@ function getStateFromProps(props: Props): State {
     const roles = props.user && props.user.roles ? props.user.roles : '';
 
     return {
+        show: true,
```
cc @devinbinnie

devinbinnie replied:
Yes we do because we will be opening the modal for a new user each time, even if it's the same user twice we always want `show` to be `true` unless we're closing the modal

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/manage_teams_modal/manage_teams_modal.tsx:
```diff
@@ -52,6 +52,10 @@ const ManageTeamsModal = ({locale, onModalDismissed, show, user, actions}: Props
         }
     }, [user?.id]);
 
+    const onModalDismissed = () => {
```
This is just a simple function i dont see value in wrapping it in callback

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.test.tsx:
```diff
@@ -2,104 +2,47 @@
-        expect(wrapper).toMatchSnapshot();
-    });
-
-    test('should match snapshot if no nickname is defined', () => {
```
implementation of this component changed as the old test became redundant

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -26,343 +22,301 @@ import ConfirmModal from 'components/confirm_modal';
+        }
+    };
+
+    getUserAuthenticationTextField(mfaEnabled: Props['mfaEnabled'], user?: UserProfile): string {
```
good idea extracted into to another function and wrote tests

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -26,343 +22,301 @@ import ConfirmModal from 'components/confirm_modal';
-    setTeamsData = (teams: TeamMembership[]): void => {
+    componentDidMount() {
+        const userId = this.props.match.params.user_id ?? '';
+        if (userId.length !== 0) {
```
Done ✅

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -26,343 +22,301 @@ import ConfirmModal from 'components/confirm_modal';
-            showPasswordModal: false,
-        });
-    };
+            await this.getUser(this.state.user.id);
```
this is needed since we are not dispatching an action

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -26,343 +22,301 @@ import ConfirmModal from 'components/confirm_modal';
-    handleResetMfa = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
-        e.preventDefault();
-        adminResetMfa(this.props.user.id, null, null);
+        this.toggleCloseModalDeactivateMember();
```
yes we need to have a ticket for this, its in discussion with UX regarding this modal changes

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -26,343 +22,301 @@ import ConfirmModal from 'components/confirm_modal';
+        this.setState({showTeamSelectorModal: false});
+    };
+
+    render() {
```
i think anymore refactor for this component could come when we move this to functional

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -379,114 +333,192 @@ class SystemUserDetail extends React.PureComponent<Props & RouteComponentProps,
-                                    <div>
-                                        <EmailIcon className='SystemUserDetail__field-icon'/>
+                                <>
+                                    <span>{this.state?.user?.position}</span>
```
yes we need the span for styling. Added empty string

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/revoke_sessions_button/index.tsx:
```diff
@@ -13,7 +13,7 @@ import {emitUserLoggedOutEvent} from 'actions/global_actions';
 import SystemPermissionGate from 'components/permissions_gates/system_permission_gate';
 
-function RevokeSessionsButton() {
+export function RevokeSessionsButton() {
```
in the system users all component were named exports so i thought why leave this poor fellow alone

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users.tsx:
```diff
@@ -1,362 +1,553 @@
-            },
-            allow_inactive: true,
-        };
+    function handleSortingChange(updateFn: (currentSortingState: SortingState) => SortingState) {
```
agree with the latter part

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -26,343 +22,301 @@ import ConfirmModal from 'components/confirm_modal';
+
+    handleTeamsLoaded = (teams: TeamMembership[]) => {
         const teamIds = teams.map((team) => team.team_id);
         this.setState({teams});
         this.setState({teamIds});
         this.setState({refreshTeams: false});
```
lets do it later when this component is converted to functional

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -26,343 +22,301 @@ import ConfirmModal from 'components/confirm_modal';
-            then((data) => this.onUpdateActiveResult(data.error));
-        this.setState({showDeactivateMemberModal: false});
-    };
+            await this.getUser(this.state.user.id);
```
this is needed as well because we are going through the redux route

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.tsx:
```diff
@@ -26,343 +22,301 @@ import ConfirmModal from 'components/confirm_modal';
-    closeAddTeam = (): void => {
-        this.setState({addTeamOpen: false});
-    };
+    handleActivateUser = async () => {
```
i think its fine as is

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users.test.tsx:
```diff
@@ -1,162 +0,0 @@
-        },
-    };
-
-    test('should match default snapshot', () => {
```
the previous test were low quality snapshot tests and the current tests are enough to cover the cases

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/index.ts:
```diff
@@ -34,57 +28,61 @@ function mapStateToProps(state: GlobalState) {
-            totalUsers = stats.total_member_count;
-        }
-    }
+    const tableProperties = getAdminConsoleUserManagementTableProperties(state);
+    const tablePropertySortColumn = tableProperties?.sortColumn ?? adminConsoleUserManagementTablePropertiesInitialState.sortColumn;
+    const tablePropertySortIsDescending = tableProperties?.sortIsDescending ?? adminConsoleUserManagementTablePropertiesInitialState.sortIsDescending;
+    const tablePropertyPageSize = tableProperties?.pageSize ?? adminConsoleUserManagementTablePropertiesInitialState.pageSize;
+    const tablePropertyPageIndex = tableProperties?.pageIndex ?? adminConsoleUserManagementTablePropertiesInitialState.pageIndex;
+    const tablePropertyCursorDirection = tableProperties?.cursorDirection ?? adminConsoleUserManagementTablePropertiesInitialState.cursorDirection;
+    const tablePropertyCursorUserId = tableProperties?.cursorUserId ?? adminConsoleUserManagementTablePropertiesInitialState.cursorUserId;
+    const tablePropertyCursorColumnValue = tableProperties?.cursorColumnValue ?? adminConsoleUserManagementTablePropertiesInitialState.cursorColumnValue;
+    const tablePropertyColumnVisibility = tableProperties?.columnVisibility ?? adminConsoleUserManagementTablePropertiesInitialState.columnVisibility;
+    const tablePropertySearchTerm = tableProperties?.searchTerm ?? adminConsoleUserManagementTablePropertiesInitialState.searchTerm;
+    const tablePropertyFilterTeam = tableProperties?.filterTeam ?? TeamFilters.AllTeams;
+    const tablePropertyFilterTeamLabel = tableProperties?.filterTeamLabel ?? '';
+    const tablePropertyFilterRole = tableProperties?.filterRole ?? RoleFilters.Any;
+    const tablePropertyFilterStatus = tableProperties?.filterStatus ?? StatusFilter.Any;
+    const tablePropertyDateRange = tableProperties?.dateRange ?? ReportDuration.AllTime;
```
gives ability to have default, but i think its fine

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users.tsx:
```diff
@@ -1,362 +1,553 @@
-            searching: false,
-        };
-    }
+function SystemUsers(props: Props) {
```
yes its fine now

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users.tsx:
```diff
@@ -1,362 +1,553 @@
-            ...teamId === SearchUserTeamFilter.NO_TEAM && {
-                [UserSearchOptions.WITHOUT_TEAM]: true,
+    const columns: Array<ColumnDef<UserReport, any>> = useMemo(
+        () => [
```
thats true, but i think its not complicated to read in my opinion since its just an object.

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users.tsx:
```diff
@@ -1,362 +1,553 @@
+                if (data.length > 0) {
+                    setUserReports(data);
+                } else {
+                    setUserReports([]);
```
Yes you are correct in event of being same they both can be same, but i like to be explicit to say we are setting it to empty

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users.tsx:
```diff
@@ -1,362 +1,553 @@
+        } as TableMeta,
+        getCoreRowModel: getCoreRowModel<UserReport>(),
+        getSortedRowModel: getSortedRowModel<UserReport>(),
+        onPaginationChange: handlePaginationChange as OnChangeFn<PaginationState>,
+        onSortingChange: handleSortingChange as OnChangeFn<SortingState>,
+        onColumnVisibilityChange: handleColumnVisibilityChange as OnChangeFn<VisibilityState>,
```
passes down to react testing library

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users.tsx:
```diff
@@ -1,362 +1,553 @@
-            return;
-        }
+    function updateUserReport(userId: string, updatedReport: Partial<UserReportWithError>) {
+        setUserReports(userReports.map((user) => {
```
i think both are correct way to that

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users.tsx:
```diff
@@ -1,362 +1,553 @@
+                        tableId={tableId}
+                        user={info.row.original}
+                        currentUser={props.currentUser}
+                        updateUser={(updatedUser) => updateUserReport(info.row.original.id, updatedUser)}
+                        onError={(error) => updateUserReport(info.row.original.id, {error})}
```
this part needs improvement as mentioned above about error handling in the modals, may be that time we can improve it, even i am not fan of inline functions

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_column_toggler_menu/index.tsx:
```diff
@@ -0,0 +1,191 @@
+export function SystemUsersColumnTogglerMenu(props: Props) {
+    const {formatMessage} = useIntl();
+
+    function getColumnName(columnId: CoreColumn<UserReport, unknown>['id']) {
```
it can but its fine as is also

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_column_toggler_menu/index.tsx:
```diff
@@ -0,0 +1,191 @@
+export function SystemUsersColumnTogglerMenu(props: Props) {
+    const {formatMessage} = useIntl();
+
+    function getColumnName(columnId: CoreColumn<UserReport, unknown>['id']) {
```
Yes thats another way to do as well

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_column_toggler_menu/index.tsx:
```diff
@@ -0,0 +1,191 @@
+                    } else {
+                        leadingElement = (
+                            <i className='icon icon-checkbox-blank-outline'/>
+                        );
+                    }
+
+                    const postStatsColumns: string[] = [ColumnNames.lastPostDate, ColumnNames.daysActive, ColumnNames.totalPosts];
+                    if (props.isMySql && postStatsColumns.includes(column.id)) {
+                        return (
+                            <WithTooltip
+                                key={column.id}
+                                id={column.id}
+                                title={formatMessage({id: 'admin.system_users.column_toggler.mysql_unavailable.title', defaultMessage: 'Not available for servers using MySQL'})}
+                                hint={formatMessage({id: 'admin.system_users.column_toggler.mysql_unavailable.desc', defaultMessage: 'Please use the export functionality to view these values'})}
+                                placement='left'
+                            >
+                                <Menu.Item
+                                    className='systemUsersColumnToggler__lockedItem'
+                                    role='menuitemcheckbox'
+                                    labels={getColumnName(column.id)}
+                                    disabled={true}
+                                    leadingElement={leadingElement}
+                                    trailingElements={<Tag text={formatMessage({id: 'admin.system_users.column_toggler.mysql_unavailable.label', defaultMessage: 'Not available'})}/>}
+                                    onClick={column.getToggleVisibilityHandler()}
+                                />
+                            </WithTooltip>
+                        );
+                    }
+
+                    return (
+                        <Menu.Item
+                            key={column.id}
+                            id={column.id}
+                            role='menuitemcheckbox'
+                            labels={getColumnName(column.id)}
+                            disabled={!column.getCanHide()}
+                            leadingElement={leadingElement}
+                            onClick={column.getToggleVisibilityHandler()}
+                        />
+                    );
```
Its actually still one component MenuItem just that addition of mySQL have added that but i am sure this one is just temporary.

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_date_range_menu/index.tsx:
```diff
@@ -0,0 +1,219 @@
+
+    const now = new Date();
+
+    function getSelectedDateRange(dateRange: ReportDuration) {
```
its fine to keep in the component as is

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_export/index.tsx:
```diff
@@ -0,0 +1,120 @@
+    dateRange: ReportDuration;
+}
+
+export function SystemUsersExport(props: Props) {
```
i am not aware that we have such convention. If there isn't i happy to leave as is.

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_export/index.tsx:
```diff
@@ -0,0 +1,120 @@
+        }
+    }
+
+    function handleExport() {
```
i think base element is button so i think i should effect

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_export/export_user_data_modal.tsx:
```diff
@@ -0,0 +1,84 @@
+        />
+    );
+
+    let message = (
+        <FormattedMessage
+            id='export_user_data_modal.dange_range.all_time'
+            defaultMessage={'You\'re about to export user data for all time. When the export is ready, a CSV file will be sent to you in a Mattermost direct message. This export will take a few minutes.'}
+        />
+    );
+    if (dateRange === ReportDuration.Last30Days) {
+        message = (
+            <FormattedMessage
+                id='export_user_data_modal.dange_range.last_30_days'
+                defaultMessage={'You\'re about to export user data for the last 30 days. When the export is ready, a CSV file will be sent to you in a Mattermost direct message. This export will take a few minutes.'}
+            />
+        );
+    } else if (dateRange === ReportDuration.PreviousMonth) {
+        message = (
+            <FormattedMessage
+                id='export_user_data_modal.dange_range.previous_month'
+                defaultMessage={'You\'re about to export user data for the previous month. When the export is ready, a CSV file will be sent to you in a Mattermost direct message. This export will take a few minutes.'}
+            />
+        );
+    } else if (dateRange === ReportDuration.Last6Months) {
+        message = (
+            <FormattedMessage
+                id='export_user_data_modal.dange_range.last_6_months'
+                defaultMessage={'You\'re about to export user data for the last 6 months. When the export is ready, a CSV file will be sent to you in a Mattermost direct message. This export will take a few minutes.'}
+            />
+        );
+    }
```
i guess its fine as is

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_filters_popover/index.tsx:
```diff
@@ -0,0 +1,169 @@
+                        {...getFloatingProps()}
+                        id='systemUsersFilterPopover'
+                        ref={floatingRefs.setFloating}
+                        style={Object.assign({}, floatingStyles, floatingTransistionStyles)}
```
its returned from hook, so it is expected to change.

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_export/upgrade_export_data_modal.tsx:
```diff
@@ -0,0 +1,62 @@
+}
+
+export function UpgradeExportDataModal({onExited}: Props) {
+    const openPricingModal = useOpenPricingModal();
+
+    const confirm = () => {
+        openPricingModal();
+    };
```
i guess it makes clear what arg are passed to the function openPricingModal.

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_filters_popover/system_users_filter_role/index.tsx:
```diff
@@ -0,0 +1,79 @@
+        ];
+    }, []);
+
+    const [value, setValue] = useState(getDefaultSelectedValueFromList(props.initialValue, options));
```
changed

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_filters_popover/system_users_filter_team/index.tsx:
```diff
@@ -0,0 +1,229 @@
+
+    const [list, setList] = useState<OptionsType<OptionType>>();
+    const [pageNumber, setPageNumber] = useState(0);
+    const [value, setValue] = useState<ValueType<OptionType>>(getDefaultSelectedTeam(props.initialValue, props.initialLabel));
```
changed

M-ZubairAhmed commented on webapp/channels/src/components/admin_console/system_users/system_users_filters_popover/system_users_filter_team/index.tsx:
```diff
@@ -0,0 +1,229 @@
+
+    function handleOnChange(value: ValueType<OptionType>) {
+        setValue(value);
+        props.onChange((value as OptionType).value as string, (value as OptionType).label as string);
```
fought with the underlying component a lot and had to do this casting

johndavidlugtu commented:
> wanted to share some initial observations, @devinbinnie @abhijit-singh ,
> 
> * In New UI, we can't see if the user is Deactivated, unless we filter them.
> 
> <img alt="Screenshot 2024-01-29 at 12 48 30 PM" width="600" src="https://private-user-images.githubusercontent.com/15979783/300364840-d6d9bbea-abe6-4260-afad-33e3333417cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNjQ4NDAtZDZkOWJiZWEtYWJlNi00MjYwLWFmYWQtMzNlMzMzMzQxN2NjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExYzljMmYwYjg5Yjc0MmE0MGMwZjdmZGVmZWViNjdmOTUyYjVlNWZlNjlmZTdiZmYwNDNkNmE4ZjVkYjczMDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hv8iHGS8mSd8AEkOZHnALNzz1MRCm3ii3d3oB2HWhOc">
> comparing to how it was earlier. <img alt="image" width="600" src="https://private-user-images.githubusercontent.com/15979783/300365044-0eabd25b-72d9-4471-9f30-e5eb5b593b88.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNjUwNDQtMGVhYmQyNWItNzJkOS00NDcxLTlmMzAtZTVlYjViNTkzYjg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyMTM5NDVlM2M3NjczYmQzOGZiM2Y5NjM4ZWMzMzhhNTkwOTYwOThiNGQzYTRiNzgwN2U1NjI1MjIxNmZlMmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4oULqXThxYKMqTg8ipr1AYm-QZCPYINhOXC_I7AmKI4">
> 
> * Any chance we can have a 'Reset filter' option (this option is available in other places like teams setting)
> 
> <img alt="image" width="716" src="https://private-user-images.githubusercontent.com/15979783/300365309-b09b6391-9f58-40d3-8d6d-98e415443e93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNjUzMDktYjA5YjYzOTEtOWY1OC00MGQzLThkNmQtOThlNDE1NDQzZTkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM0MTA2YTNiNzE5ZmVjYTk4ZGQ2YjM5MDM1OWU3ZGFhMTNkM2NlNmQ2NjIzZjc0OWUyYzZmZTQ4MGE3ZGYwMTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1JMqsAVoN-J5baIfTBE9oAIQ2R1JcH0ALIjA5PaOOsg">
> * Should the Filter option be `Activated User`. there was a recent change in team & site statistics around the usage of words Active-Activated user @larkox @johndavidlugtu
> 
> <img alt="image" width="261" src="https://private-user-images.githubusercontent.com/15979783/300378536-f6f5fe8b-ab7c-470d-9c61-f039c1ccc5fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNzg1MzYtZjZmNWZlOGItYWI3Yy00NzBkLTljNjEtZjAzOWMxY2NjNWZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjYzZmMWM2MjFmYTU4ZDI2YzQ0NDIwYTQ1MWNlMGY4ZTNmNjk0MDg2MzI1OTk4ZTRmZDU5MjU2YTJkMTUzNjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._bYG6sonufh7-eb8UJdHvIg3N0OyTWDNiGCD5HNmZ6k">
> * a small spelling correction needed `Manage Roles`. Also should it be like Camel casing?
> 
> <img alt="image" width="167" src="https://private-user-images.githubusercontent.com/15979783/300427033-ca542d24-51ab-4b6c-a6e8-eb4a6b192e9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDA0MjcwMzMtY2E1NDJkMjQtNTFhYi00YjZjLWE2ZTgtZWI0YTZiMTkyZTlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3ZTM3ZThjNGZlNzAyNWY1OWZjZGVjZDQ5MDBmMjNmMWRjMTI2YzE3ZDk1ZDU3NzU5ZDU1NTNiMmZlMjY0ZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._BkIFMrKlRExOQ3OZ7NeFgVXaFxkKTJ-rZXpzsoTDXk">
> (as seen in 9.5) <img alt="image" width="304" src="https://private-user-images.githubusercontent.com/15979783/300427564-77f95548-ea22-4c68-84bd-edd23f9abeeb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDA0Mjc1NjQtNzdmOTU1NDgtZWEyMi00YzY4LTg0YmQtZWRkMjNmOWFiZWViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1YTdjMzUxZjE3NjVhYWI0ZjViYzRhN2IxNGU1MTdhOTgwNjQ3ZTNhODZiMGEyOTg4YTcyYzA1ZmM5ZGYzMmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8gTshGOLrTiN2RqEmApPiJaSlyB1xbPTlEAeWTMZnqI">
> 
> * if term `User` is searched in console, the column header in user manager splits
> 
> <img alt="Screenshot 2024-01-29 at 1 28 28 PM" width="560" src="https://private-user-images.githubusercontent.com/15979783/300375192-b10ec889-d57c-4c72-90a5-87b401431d70.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNzUxOTItYjEwZWM4ODktZDU3Yy00YzcyLTkwYTUtODdiNDAxNDMxZDcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzYmY1YTkxY2UzNTU1YmE4NTkwMGY0MjU1NDUwODc1YzEzZDNmNGNmYzYxMGE5YjgwYjI2MWUwODU1MjE5ZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xwhF7fBfsUMLITwXcMVmE1gvnoE8bDulyyDBqy5spkc">
> * If we have the above search term, and apply Role filters.
> 
> <img alt="Screenshot 2024-01-29 at 1 29 05 PM" width="736" src="https://private-user-images.githubusercontent.com/15979783/300375370-b8becb92-308c-48a7-8c51-f98a912d75c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNzUzNzAtYjhiZWNiOTItMzA4Yy00OGE3LThjNTEtZjk4YTkxMmQ3NWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyNWQ1MmNhMzU2ZGYzNWFlYzE5NmEyZDJlN2EzMjgyYjc4ZGQ4Nzk5NzYzYmQ4MzczZjQzODlmMzllNjE2NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XrtHZ4bW4YZRDlMRIWk_oEFwOvCaSUKarK1BKMkFvdQ"> <img alt="image" width="579" src="https://private-user-images.githubusercontent.com/15979783/300377095-5c6f96b7-abcf-4961-8745-3b6386830fe5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNzcwOTUtNWM2Zjk2YjctYWJjZi00OTYxLTg3NDUtM2I2Mzg2ODMwZmU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlYzVmZTFiZmI2ZDJmYTUzZDgwNDcwY2JjMzFmZmJlYzBmMzU4NDJlNjE1Mzg2ZWMxMDk5OTU4ODZiYTc0MGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.51X7aTPGjjmuFeCzta9HEMwPMJa5LugzIOE80sax-Wc">
> * some e2e might need locators to be updated.
> * The Last post/last active, were empty. Had logged in with another user to post in channel
> 
> <img alt="image" width="800" src="https://private-user-images.githubusercontent.com/15979783/300366326-ae3c7cda-daf5-4c05-b388-c7e028711604.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNjYzMjYtYWUzYzdjZGEtZGFmNS00YzA1LWIzODgtYzdlMDI4NzExNjA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiMTg3YjdjM2MyZDk3NGFhZWYyZmY3ODBlZDJjNDBiODk4OTZkYjgyMTdjNjNhYWFmZTAwMjhiYWViNWVhNjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BkKQeqhRDtvQZTzTe2n3HZzuG6drWI3zkRT2xQUxhEc">
> * filters are working as expected ✅
> * sorting, pagination ✅
> * export data ✅
> 
> <img alt="image" width="700" src="https://private-user-images.githubusercontent.com/15979783/300367415-bb43f419-e67e-4b3e-9a8f-93040a3bdee5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MjgzMDYsIm5iZiI6MTcwNjcyODAwNiwicGF0aCI6Ii8xNTk3OTc4My8zMDAzNjc0MTUtYmI0M2Y0MTktZTY3ZS00YjNlLTlhOGYtOTMwNDBhM2JkZWU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4NDUyYmRjZGViZGRmZmY4NzdmMDFiODY3OTYzMGYyYWIyYmE5N2VjOTRmNjA3NzIxN2ZjZjBiNzAwNGNlOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CQCLFODp1ZEZwoR01wexZGeV9ySS7pXeR9pAfuhdoc8">
> * selected Columns being displayed ✅
> * revoke sessions ✅

+1 we should use the new "Activated" terminology to refer to users that have not been deactivated

yasserfaraazkhan commented:
/e2e-tests

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](http://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/406801) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/406801)

M-ZubairAhmed commented:
@yasserfaraazkhan are all e2e good now?

yasserfaraazkhan commented:
> @yasserfaraazkhan are all e2e good now?

@M-ZubairAhmed , just these on CI failed.
will try and fix these.
`enterprise/system_console/authentication_method_spec.js`, `system_console/user_management_spec.js`

yasserfaraazkhan commented:
/e2e-tests

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](http://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/406902) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/406902)

M-ZubairAhmed commented:
@yasserfaraazkhan i was wondering if we can just delete the related cypress e2e and write them in playwright instead of fixing these? what do you think

yasserfaraazkhan commented:
> @yasserfaraazkhan i was wondering if we can just delete the related cypress e2e and write them in playwright instead of fixing these? what do you think

Sure @M-ZubairAhmed , playwright test would be great.
I've fix most of them. Untill the new playwright, these will be helpful running in master.

devinbinnie commented on server/channels/app/report.go:
```diff
@@ -64,22 +64,24 @@ func (a *App) compileCSVChunks(prefix string, numberOfChunks int, headers []stri
 			return err
 		}
-		if _, err = a.AppendFile(bytes.NewReader(chunk), filePath); err != nil {
+		_, writeErr := compiledBuf.Write(chunk)
```
`AppendFile` was written assuming that we were uploading large potentially multipart files, and is designed to work on bigger parts. Our S3 library enforces a minimum part size of 5MB, so since most of our reports for the time being are less than that, I think we're safe to remove this very slight optimization.

larkox reviewed (approved):
Still some style concerns, but let's not block this PR. We will discuss the style concerns offline.

mattermost-build commented:
E2E tests not automatically triggered, because the PR is not in a mergeable state. Please update the branch with the base branch and resolve outstanding conflicts.

yasserfaraazkhan reviewed (approved):
[the search defect is fixed](https://mattermost.atlassian.net/browse/MM-56728) ✅

abhijit-singh reviewed (approved):
This is brilliant work @devinbinnie and @M-ZubairAhmed! Love it! 🎉 

I see that a couple of @yasserfaraazkhan's points mentioned [in this comment](https://github.com/mattermost/mattermost/pull/26055#pullrequestreview-1857188528) have not been fixed yet, but other than that looks great to me! 

I have a couple of follow up tasks for design around this - better visuals for errors ([MM-56720](https://mattermost.atlassian.net/jira/software/c/projects/MM/issues/MM-56720)) and better empty states, that we can queue up as follow up or even help-wanted tickets once design-ready.
